api/integration/registry/license_providers.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from typing import Type, cast
+
+from api.circulation import BaseCirculationAPI
+from core.integration.goals import Goals
+from core.integration.registry import IntegrationRegistry
+from core.model.configuration import ExternalIntegration
+
+
+class LicenseProvidersRegistry(IntegrationRegistry["BaseCirculationAPI"]):
+    def __init__(self) -> None:
+        super().__init__(Goals.LICENSE_GOAL)
+
+        from api.axis import Axis360API
+        from api.bibliotheca import BibliothecaAPI
+        from api.enki import EnkiAPI
+        from api.lcp.collection import LCPAPI
+        from api.odilo import OdiloAPI
+        from api.odl import ODLAPI, SharedODLAPI
+        from api.odl2 import ODL2API
+        from api.opds_for_distributors import OPDSForDistributorsAPI
+        from api.overdrive import OverdriveAPI
+        from core.opds2_import import OPDS2Importer
+        from core.opds_import import OPDSImporter
+
+        apis = {
+            ExternalIntegration.OVERDRIVE: OverdriveAPI,
+            ExternalIntegration.ODILO: OdiloAPI,
+            ExternalIntegration.BIBLIOTHECA: BibliothecaAPI,
+            ExternalIntegration.AXIS_360: Axis360API,
+            EnkiAPI.ENKI_EXTERNAL: EnkiAPI,
+            OPDSForDistributorsAPI.NAME: OPDSForDistributorsAPI,
+            ODLAPI.NAME: ODLAPI,
+            ODL2API.NAME: ODL2API,
+            SharedODLAPI.NAME: SharedODLAPI,
+            LCPAPI.NAME: LCPAPI,
+            OPDSImporter.NAME: OPDSImporter,
+            OPDS2Importer.NAME: OPDS2Importer,
```

Comment:
The bottom 2 aren't technically BaseCirculationAPIs but behave close enough to them. Hence the `cast` below.

Reply:
Implemented a Protocol class. I think mypy was introduced by the BDFL to prove no dictator is truly benevolent.